### ISSUE TITLE
datetime column error "conversion of a nvarchar data type to a datetime data type resulted in an out-of-range value"

### ISSUE THREAD

**Issue by fun-ruby** [JesseChavez/activerecord-jdbc-adapter#11]
I'm using Rails 5.2.3 with JRuby 9.2.8 on a Rails 4.2 database schema, so `datetime` data type is being used.

Should also mention using version 52.2.3 of this gem and jdbc-mssql 0.6.0.

Added the `time_zone_aware_types` line to application.rb but see no difference in outcome. In particular, the `datetime` columns are getting `datetime2` data format on insert and update.

`["created_at", 2019-11-25 15:54:53 UTC], ["updated_at", 2019-11-25 15:54:53 UTC]`

Would be grateful for suggestion to resolve this.

**Comment by JesseChavez** [comment]
Hi There,

Can you post the whole error you are getting?, also would be great to know the version of the
`activerecord-jdbc-alt-adapter` of the gem you are using.

**Comment by JesseChavez** [comment]
FYI:

The SQL Server `datetime` data type is mapped as `datatime_basic` in activerecord, and  `datetime2` to the standard activerecord `datatime` . I've done this way because I needed more precision in `datetime` in my apps. 

`datetime_basic` will work for simple updates and inserts as you said before, for more complex queries such are `where`, `update_all` you might have issues.

The possible solutions are:

**1) change the code, for example if you are doing `where` do something like:**

```
Model.where('created_at < ? ', Time.current.to_s(:db))
```
The idea is to pass an string formatted in the accepted way by SQL Server `datetime` type

**2) Write a simple migration to convert `datetime` to `datetime2`  (preferred way)**

example migration:
```ruby
class FixMiscDatetimeFieldsPartOne < ActiveRecord::Migration[5.0]
  def up
    if column_exists?(:insurance_products, :expiry_date)
      remove_index(:insurance_products, :expiry_date) if index_exists?(:insurance_products, :expiry_date)
      change_column :insurance_products, :expiry_date, :datetime
    end
  end

  def down
  end
end
```
that is snippet of a migration I did long ago for one of the apps I maintain at work (around May 2019 when I was upgrading to rails 5.0)

**Comment by fun-ruby** [comment]
Thanks for looking at this, @JesseChavez. Above suggestions are possible work-around that I can certainly use, with change_column being a better option.

My original error stems from ActiveRecord injected timestamps, such as on Insert or Update. There may be a way to override the date format but I haven't look too far.

I will post the stack trace next.

**Comment by fun-ruby** [comment]
Below is the top portion of the stack trace. I can post the rest of the `org.jruby.*` trace if you'd like but it is really long.

```
ActiveRecord::StatementInvalid: ActiveRecord::JDBCError: com.microsoft.sqlserver.jdbc.SQLServerException: The conversion of a nvarchar data type to a datetime data type resulted in an out-of-range value.: INSERT INTO [addresses] ([field01], [field02], [field03], [address], [city], [state], [country], [field08], [field09], [field10], [created_at], [updated_at]) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
arjdbc/jdbc/RubyJdbcConnection.java:922:in `execute_insert'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-jdbc-alt-adapter-52.2.3-java/lib/arjdbc/abstract/database_statements.rb:17:in `block in exec_insert'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract_adapter.rb:581:in `block in log'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract_adapter.rb:580:in `block in log'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activesupport-5.2.3/lib/active_support/notifications/instrumenter.rb:23:in `instrument'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract_adapter.rb:571:in `log'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-jdbc-alt-adapter-52.2.3-java/lib/arjdbc/abstract/core.rb:72:in `log'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-jdbc-alt-adapter-52.2.3-java/lib/arjdbc/abstract/database_statements.rb:16:in `exec_insert'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-jdbc-alt-adapter-52.2.3-java/lib/arjdbc/mssql/database_statements.rb:51:in `exec_insert'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/database_statements.rb:162:in `insert'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/query_cache.rb:21:in `insert'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-jdbc-alt-adapter-52.2.3-java/lib/arjdbc/abstract/database_statements.rb:48:in `insert'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/persistence.rb:187:in `_insert_record'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/persistence.rb:734:in `_create_record'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/counter_cache.rb:184:in `_create_record'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/locking/optimistic.rb:70:in `_create_record'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/attribute_methods/dirty.rb:140:in `_create_record'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/callbacks.rb:346:in `block in _create_record'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activesupport-5.2.3/lib/active_support/callbacks.rb:132:in `run_callbacks'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activesupport-5.2.3/lib/active_support/callbacks.rb:816:in `_run_create_callbacks'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/callbacks.rb:346:in `_create_record'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/timestamp.rb:102:in `_create_record'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/persistence.rb:705:in `create_or_update'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/callbacks.rb:342:in `block in create_or_update'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activesupport-5.2.3/lib/active_support/callbacks.rb:132:in `run_callbacks'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activesupport-5.2.3/lib/active_support/callbacks.rb:816:in `_run_save_callbacks'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/callbacks.rb:342:in `create_or_update'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/persistence.rb:308:in `save!'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/validations.rb:52:in `save!'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/transactions.rb:315:in `block in save!'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/transactions.rb:387:in `block in with_transaction_returning_status'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/database_statements.rb:265:in `transaction'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/transactions.rb:212:in `transaction'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/transactions.rb:385:in `with_transaction_returning_status'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/transactions.rb:315:in `save!'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/suppressor.rb:48:in `save!'
/home/long/rails5/newapp/lib/dev_setup_profile.rb:63:in `create_address'
/home/long/rails5/newapp/db/abstract_seeder.rb:51:in `block in create_user_address'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/database_statements.rb:267:in `block in transaction'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/transaction.rb:239:in `block in within_new_transaction'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/transaction.rb:236:in `within_new_transaction'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/database_statements.rb:267:in `transaction'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/transactions.rb:212:in `transaction'
/home/long/rails5/newapp/db/abstract_seeder.rb:48:in `create_user_address'
/home/long/rails5/newapp/db/seeds_user.rb:44:in `seed_user_address'
/home/long/rails5/newapp/db/abstract_seeder.rb:13:in `run'
/home/long/rails5/newapp/db/seeds_user.rb:17:in `run'
/home/long/rails5/newapp/db/seeds.rb:69:in `<main>'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:1:in `block in (root)'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:257:in `load_dependency'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:285:in `load'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:285:in `block in load'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/railties-5.2.3/lib/rails/engine.rb:551:in `load_seed'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/railties-5.2.3/lib/rails/engine.rb:666:in `with_inline_jobs'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/railties-5.2.3/lib/rails/engine.rb:551:in `load_seed'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/tasks/database_tasks.rb:281:in `block in load_seed'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/railties/databases.rake:194:in `block in <main>'

Caused by:
ActiveRecord::JDBCError: com.microsoft.sqlserver.jdbc.SQLServerException: The conversion of a nvarchar data type to a datetime data type resulted in an out-of-range value.
arjdbc/jdbc/RubyJdbcConnection.java:922:in `execute_insert'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-jdbc-alt-adapter-52.2.3-java/lib/arjdbc/abstract/database_statements.rb:17:in `block in exec_insert'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract_adapter.rb:581:in `block in log'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract_adapter.rb:580:in `block in log'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activesupport-5.2.3/lib/active_support/notifications/instrumenter.rb:23:in `instrument'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract_adapter.rb:571:in `log'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-jdbc-alt-adapter-52.2.3-java/lib/arjdbc/abstract/core.rb:72:in `log'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-jdbc-alt-adapter-52.2.3-java/lib/arjdbc/abstract/database_statements.rb:16:in `exec_insert'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-jdbc-alt-adapter-52.2.3-java/lib/arjdbc/mssql/database_statements.rb:51:in `exec_insert'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/database_statements.rb:162:in `insert'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/query_cache.rb:21:in `insert'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-jdbc-alt-adapter-52.2.3-java/lib/arjdbc/abstract/database_statements.rb:48:in `insert'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/persistence.rb:187:in `_insert_record'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/persistence.rb:734:in `_create_record'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/counter_cache.rb:184:in `_create_record'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/locking/optimistic.rb:70:in `_create_record'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/attribute_methods/dirty.rb:140:in `_create_record'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/callbacks.rb:346:in `block in _create_record'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activesupport-5.2.3/lib/active_support/callbacks.rb:132:in `run_callbacks'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activesupport-5.2.3/lib/active_support/callbacks.rb:816:in `_run_create_callbacks'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/callbacks.rb:346:in `_create_record'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/timestamp.rb:102:in `_create_record'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/persistence.rb:705:in `create_or_update'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/callbacks.rb:342:in `block in create_or_update'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activesupport-5.2.3/lib/active_support/callbacks.rb:132:in `run_callbacks'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activesupport-5.2.3/lib/active_support/callbacks.rb:816:in `_run_save_callbacks'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/callbacks.rb:342:in `create_or_update'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/persistence.rb:308:in `save!'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/validations.rb:52:in `save!'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/transactions.rb:315:in `block in save!'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/transactions.rb:387:in `block in with_transaction_returning_status'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/database_statements.rb:265:in `transaction'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/transactions.rb:212:in `transaction'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/transactions.rb:385:in `with_transaction_returning_status'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/transactions.rb:315:in `save!'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/suppressor.rb:48:in `save!'
/home/long/rails5/newapp/lib/dev_setup_profile.rb:63:in `create_address'
/home/long/rails5/newapp/db/abstract_seeder.rb:51:in `block in create_user_address'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/database_statements.rb:267:in `block in transaction'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/transaction.rb:239:in `block in within_new_transaction'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/transaction.rb:236:in `within_new_transaction'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/connection_adapters/abstract/database_statements.rb:267:in `transaction'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/transactions.rb:212:in `transaction'
/home/long/rails5/newapp/db/abstract_seeder.rb:48:in `create_user_address'
/home/long/rails5/newapp/db/seeds_user.rb:44:in `seed_user_address'
/home/long/rails5/newapp/db/abstract_seeder.rb:13:in `run'
/home/long/rails5/newapp/db/seeds_user.rb:17:in `run'
/home/long/rails5/newapp/db/seeds.rb:69:in `<main>'

/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:1:in `block in (root)'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:257:in `load_dependency'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:285:in `load'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activesupport-5.2.3/lib/active_support/dependencies.rb:285:in `block in load'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/railties-5.2.3/lib/rails/engine.rb:551:in `load_seed'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/railties-5.2.3/lib/rails/engine.rb:666:in `with_inline_jobs'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/railties-5.2.3/lib/rails/engine.rb:551:in `load_seed'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/tasks/database_tasks.rb:281:in `block in load_seed'
/home/long/.rvm/gems/jruby-9.2.8.0@rails5203/gems/activerecord-5.2.3/lib/active_record/railties/databases.rake:194:in `block in <main>'

Caused by:
Java::ComMicrosoftSqlserverJdbc::SQLServerException: The conversion of a nvarchar data type to a datetime data type resulted in an out-of-range value.
com.microsoft.sqlserver.jdbc.SQLServerException.makeFromDatabaseError(SQLServerException.java:262)
com.microsoft.sqlserver.jdbc.SQLServerStatement.getNextResult(SQLServerStatement.java:1621)
com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement.doExecutePreparedStatement(SQLServerPreparedStatement.java:592)
com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement$PrepStmtExecCmd.doExecute(SQLServerPreparedStatement.java:522)
com.microsoft.sqlserver.jdbc.TDSCommand.execute(IOBuffer.java:7194)
com.microsoft.sqlserver.jdbc.SQLServerConnection.executeCommand(SQLServerConnection.java:2935)
com.microsoft.sqlserver.jdbc.SQLServerStatement.executeCommand(SQLServerStatement.java:248)
com.microsoft.sqlserver.jdbc.SQLServerStatement.executeStatement(SQLServerStatement.java:223)
com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement.executeUpdate(SQLServerPreparedStatement.java:471)
arjdbc.jdbc.RubyJdbcConnection$11.call(RubyJdbcConnection.java:930)
arjdbc.jdbc.RubyJdbcConnection$11.call(RubyJdbcConnection.java:922)
arjdbc.jdbc.RubyJdbcConnection.withConnection(RubyJdbcConnection.java:3477)
arjdbc.jdbc.RubyJdbcConnection.withConnection(RubyJdbcConnection.java:3443)
arjdbc.jdbc.RubyJdbcConnection.execute_insert(RubyJdbcConnection.java:922)
arjdbc.jdbc.RubyJdbcConnection$INVOKER$i$execute_insert.call(RubyJdbcConnection$INVOKER$i$execute_insert.gen)
org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:396)
org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:205)
org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:325)
org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72)
org.jruby.ir.interpreter.Interpreter.INTERPRET_BLOCK(Interpreter.java:114)
org.jruby.runtime.MixedModeIRBlockBody.commonYieldPath(MixedModeIRBlockBody.java:151)
org.jruby.runtime.IRBlockBody.yieldSpecific(IRBlockBody.java:89)
org.jruby.runtime.Block.yieldSpecific(Block.java:139)
org.jruby.ir.runtime.IRRuntimeHelpers.yieldSpecific(IRRuntimeHelpers.java:481)
org.jruby.ir.instructions.YieldInstr.interpret(YieldInstr.java:76)
org.jruby.ir.interpreter.StartupInterpreterEngine.processOtherOp(StartupInterpreterEngine.java:178)
org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:104)
org.jruby.ir.interpreter.Interpreter.INTERPRET_BLOCK(Interpreter.java:114)
org.jruby.runtime.MixedModeIRBlockBody.commonYieldPath(MixedModeIRBlockBody.java:151)
org.jruby.runtime.IRBlockBody.yieldSpecific(IRBlockBody.java:89)
org.jruby.runtime.Block.yieldSpecific(Block.java:139)
org.jruby.ir.targets.YieldSite.yieldSpecific(YieldSite.java:141)
...
```

**Comment by fun-ruby** [comment]
Also, I am using `activerecord-jdbc-alt-adapter 52.2.3` and `jdbc-mssql 0.6.0`

**Comment by JesseChavez** [comment]
Here is  sample app I created a while ago, I changed the timestamps fields from `datatime2` to `datetime`

https://github.com/JesseChavez/mssql_sam52

but everything seems to work fine which lead me to think that there is something wrong  with the `datatime` formatting  in you SQL server or you rails app default time formatting.


Here is some info regarding the datime formatting.

https://docs.microsoft.com/en-us/sql/t-sql/data-types/datetime-transact-sql?view=sql-server-ver15#ansi-and-iso-8601-compliance

Do you have any default `datetime` formating in your rails app ?

something like https://thoughtbot.com/blog/custom-formats-for-datetime

It is not recommend to change the default `datetime` format, better is to create other for your use case or do the formatting in `I18n`

**Comment by fun-ruby** [comment]
Thanks for the sample app and reference links. I had removed default `datetime` format before getting to this issue.

I think my expectation on how things work may have been flawed. When testing against a new database (created from `rake db:migrate`), all unit tests passed. Migration scripts declared as `t.datetime` mapped to DB `datetime2`.

Next I tested against an existing database which was created in Rails 4, where `t.datetime` mapped to DB `datetime`. I had expected things to just work as is. Now I think the best option moving forward is to modify the existing database and change all old `datetime` columns to `datetime2`, which you had recommended before.

If you'll agree, we can close this issue.

Thanks again for all your help. :-D
